master/flock_server/static/flock-mpi.js

```diff
@@ -508,3 +510,50 @@ mpi.ireduce = async function (sendArr, op, comm) {
         return local;
     }
 }
+
+
+/**
+ * Each process sends the contents of its sendbuffer to the root process. The root process
+ * receives the messages and stores them in rank order.
+ * 
+ * @async
+ * @function mpi.igather
+ * 
+ * @param {array} sendArr - Array to 
+ * @param {function} root - The rank of the root node for the gather operation
+ * @param {string} comm - Name of the communication group to operate under
+ * 
+ * @returns {serializable} - Result of the gather is given to the node with rank === 0, other nodes will receive error codes
+ * 
+ */
+mpi.igather = async function (sendArr, root, comm) {
+
+    const TAG = 'internal_igather';
+        
+    let size = await mpi.getSize(comm);
+    let rank = await mpi.getRank(comm);
+    
+    if (rank === root) {
+
+        if (!Array.isArray(sendArr)) {
+            console.error("Argument sendArr in scatter must be an array");
+            return;
+        }
+        
+        for (let i = 1; i < size; i++) {
```

Comment:
I know we do this elsewhere in the codebase, but what the use of `await` is going to do is sequentialize the creation of requests so that we do not create the promise for each rank until the one before it resolves. This may be alright since the node will receive the message anyways and then the promise should resolve immediately when `irecv` is called for later promises though.

Reply:
I think we need to await in order so that we concat the right way.